modules/dop/services/workbench/workbench.go

```diff
@@ -0,0 +1,194 @@
+func (w *Workbench) GetUndoneProjectItem(userID string, issueSize int, pro apistructs.ProjectDTO) (*apistructs.WorkbenchProjectItem, error) {
+	var issueItem apistructs.WorkbenchProjectItem
+	issueItem.IssueList = make([]apistructs.Issue, 0)
+	issueReq := apistructs.IssuePagingRequest{
+		OrgID:    int64(pro.OrgID),
+		PageNo:   1,
+		PageSize: uint64(issueSize),
+		IssueListRequest: apistructs.IssueListRequest{
+			ProjectID:    uint64(pro.ID),
+			StateBelongs: StateBelongs,
+			Assignees:    []string{userID},
+			External:     true,
+			OrderBy:      "plan_finished_at asc, FIELD(priority, 'URGENT', 'HIGH', 'NORMAL', 'LOW')",
+			Priority:     IssuePriorities,
+			Type:         IssueTypes,
+			Asc:          true,
+		},
+	}
+	issues, total, err := w.issueSvc.Paging(issueReq)
+	if err != nil {
+		return nil, err
+	}
+	issueItem.TotalIssueNum = int(total)
+	issueItem.IssueList = issues
+	issueItem.ProjectDTO = pro
+	return &issueItem, nil
+}
+
+// SetSpecialIssueNum concurrent query different expire issue num
+func (w *Workbench) SetSpecialIssueNum(req apistructs.IssuePagingRequest, items []*apistructs.WorkbenchProjectItem) error {
+	if len(items) == 0 {
+		return nil
+	}
+	var projectIDS []uint64
+	for _, item := range items {
+		projectIDS = append(projectIDS, item.ProjectDTO.ID)
+	}
+	nowTime := time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(), 0, 0, 0, 0, time.Now().Location())
+	tomorrow := nowTime.Add(time.Hour * time.Duration(24))
+	twoDay := nowTime.Add(time.Hour * time.Duration(24*2))
+	sevenDay := nowTime.Add(time.Hour * time.Duration(24*7))
+	thirtyDay := nowTime.Add(time.Hour * time.Duration(24*30))
+	timeList := [][]int64{
+		{0, 0}, // not specified
+		{1, nowTime.Add(time.Second * time.Duration(-1)).Unix()},                 // expired
+		{nowTime.Unix(), tomorrow.Add(time.Second * time.Duration(-1)).Unix()},   // today expired
+		{tomorrow.Unix(), twoDay.Add(time.Second * time.Duration(-1)).Unix()},    // tomorrow expired
+		{twoDay.Unix(), sevenDay.Add(time.Second * time.Duration(-1)).Unix()},    // seven day expired
+		{sevenDay.Unix(), thirtyDay.Add(time.Second * time.Duration(-1)).Unix()}, // thirty day expired
+		{thirtyDay.Unix(), 0}, //feature expired
+	}
+
+	var wg sync.WaitGroup
+	var iErr error
+	wg.Add(len(expireDays))
+	for index, et := range expireDays {
+		go func(idx int, ed string) {
+			defer wg.Done()
+			etIssueReq := apistructs.IssuePagingRequest{}
+			etIssueReq.State = req.State
+			etIssueReq.StartFinishedAt = timeList[idx][0] * 1000
+			if ed == issueUnspecified {
+				etIssueReq.IsEmptyPlanFinishedAt = true
+			} else {
+				if timeList[idx][1] != 0 {
+					etIssueReq.EndFinishedAt = timeList[idx][1] * 1000
+				}
+			}
+			etIssueReq.StateBelongs = StateBelongs
+			etIssueReq.External = true
+			etIssueReq.Type = IssueTypes
+			etIssueReq.Assignees = req.Assignees
+			prosIssueNumList, err := w.issueSvc.GetSpecialIssueNum(projectIDS, etIssueReq)
```

Comment:
move the selected code into a new func

Reply:
the new func is simalar with w.issueSvc.GetSpecialIssueNum(), and new func need to custom timeList and projectIDS again

Reply:
ok, your opinion makes sense, so be it.